docs/update-empty-customer-data-from-addresses/script.liquid

```diff
@@ -0,0 +1,174 @@
+              phone
+            }
+            addresses {
+              firstName
+              lastName
+              phone
+            }
+          }
+        }
+      }
+    }
+  {% endcapture %}
+
+  {% action "shopify" %}
+    mutation {
+      bulkOperationRunQuery(
+        query: {{ bulk_operation_query | json }}
+      ) {
+        bulkOperation {
+          id
+          status
+        }
+        userErrors {
+          field
+          message
+        }
+      }
+    }
+  {% endaction %}
+
+{% elsif event.topic == "mechanic/shopify/bulk_operation" %}
+  {% assign customers = bulkOperation.objects %}
+{% endif %}
+
+{% if event.preview %}
+  {% capture customers_json %}
+    [
+      {
+        "id": "gid://shopify/Customer/1234567890",
+        "phone": null,
+        "defaultAddress": {
+          "firstName": "Jean",
+          "lastName": "Deaux"
+        },
+        "addresses": [
+          {
+            "phone": null
+          },
+          {
+            "phone": "+18885559876"
+          }
+        ]
+      }
+    ]
+  {% endcapture %}
+
+  {% assign customers = customers_json | parse_json %}
+{% endif %}
+
+{% for customer in customers %}
+  {% assign first_name = nil %}
+  {% assign last_name = nil %}
+  {% assign phone = nil %}
+
+  {% assign customer_addresses
+    = customer.addresses
+    | unshift: customer.defaultAddress
+  %}
+
+  {% if update_customer_first_and_last_name %}
+    {% if customer.firstName == blank %}
+      {% for customer_address in customer_addresses %}
+        {% unless customer_address.firstName == blank %}
+          {% assign first_name = customer_address.firstName %}
+          {% break %}
+        {% endunless %}
+      {% endfor %}
+    {% endif %}
+
+    {% if customer.lastName == blank %}
+      {% for customer_address in customer_addresses %}
+        {% unless customer_address.lastName == blank %}
+          {% assign last_name = customer_address.lastName %}
+          {% break %}
+        {% endunless %}
+      {% endfor %}
+    {% endif %}
+  {% endif %}
+
+  {% if update_customer_phone %}
+    {% if customer.phone == blank %}
+      {% for customer_address in customer_addresses %}
+        {% unless customer_address.phone == blank %}
+          {% assign phone = customer_address.phone %}
+          {% break %}
+        {% endunless %}
+      {% endfor %}
+    {% endif %}
+  {% endif %}
+
+  {% if first_name or last_name or phone %}
+    {% action "shopify" %}
+      mutation {
+        customerUpdate(
+          input: {
+            id: {{ customer.id | json }}
+            {% if first_name %}
+              firstName: {{ first_name | json }}
+            {% endif %}
+            {% if last_name %}
+              lastName: {{ last_name | json }}
+            {% endif %}
+            {% if phone %}
+              phone: {{ phone | json }}
+            {% endif %}
+          }
+        ) {
+          userErrors {
+            field
+            message
+          }
```

Comment:
Shopify _sometimes_ tweaks values on their way in. I'm not sure if this applies here or not, but I _think_ it's still a good idea to have Shopify echo back the values that they've decided to store.

Haven't tested the below, but I think this'll work:

```suggestion
          }
          customer {
            id
            firstName
            lastName
            phone
          }
```

Reply:
Ahh, good idea. I _think_ that Shopify might have already altered the phones in the addresses to be E.164 for example, but maybe not for manual entries. So, yep, I can do this, as it will certainly help on any log reviews by support or merchant.